### ISSUE TITLE
Make App-Domains Configurable

### ISSUE THREAD

**Issue by Flask** [SUSE/scf#1856]
Hello guys,

at the moment fissile sets for the apps the same domain as for the system:
```
properties.app_domains: '["((DOMAIN))"]'
properties.domain: '"((DOMAIN))"'
```
that means, that when the api pods get recreated, they automatically re-register the domain as shared domain. 
If you have a look at the source of the [cloud_controller](https://github.com/cloudfoundry/cloud_controller_ng/blob/db98f1022f0af063eb54e0ab6ad1a41231a7f487/lib/cloud_controller/seeds.rb#L71) it would make sense to separate app and system domain.

Thanks for your feedback.

**Comment by viovanov** [comment]
@Flask is it possible to reconfigure this after installation?
e.g. delete the shared domain and create another just for apps?

**Comment by Flask** [comment]
@viovanov of course you can delete the "default" shared domain, but the issue is that this domain is going to be recreated as soon as an api-pod gets started.

Expected behaviour would be that once you delete the domain, it would not be created again.

**Comment by jandubois** [comment]
Hi @Flask,

with the current `HEAD` of the `develop` branch, using a very recent fissile version, it is possible to overwrite any job property from the `values.yaml` file in the helm chart.

I tested this using

```
bosh:
  instance_groups:
  - name: api-group
    jobs:
    - name: cloud_controller_ng
      properties:
        app_domains:
        - 192.168.77.77.nip.io
```

in a Vagrant box dev setup. All the regular URLs use the cf-dev.io domain, but apps will get 192.168.77.77.nip.io domains:

```
$ cf curl /v2/info
{
   "name": "SCF",
   "build": "2.0.2",
   "support": "support@example.com",
   "version": 2,
   "description": "SUSE Cloud Foundry",
   "authorization_endpoint": "https://cf.uaa.cf-dev.io:2793",
   "token_endpoint": "https://cf.uaa.cf-dev.io:2793",
   "min_cli_version": null,
   "min_recommended_cli_version": null,
   "api_version": "2.115.0",
   "app_ssh_endpoint": "ssh.cf-dev.io:2222",
   "app_ssh_host_key_fingerprint": "ae:ca:b7:2d:bb:f4:bc:ba:7c:31:89:a2:df:94:14:53",
   "app_ssh_oauth_client": "ssh-proxy",
   "doppler_logging_endpoint": "wss://doppler.cf-dev.io:4443",
   "routing_endpoint": "https://api.cf-dev.io/routing",
   "user": "39518eea-a61a-4dea-b2cc-2715ad00e0c8"
}

$ cf push
Pushing from manifest to org suse / space myspace as admin...
Using manifest file /Users/jan/hpcloud/Stackato-Apps/phpinfo/manifest.yml
Getting app info...
Creating app with these attributes...
+ name:       phpinfo
  path:       /Users/jan/hpcloud/Stackato-Apps/phpinfo
+ memory:     64M
  routes:
+   phpinfo.192.168.77.77.nip.io
```

It is not the same as having a proper setting for it, but it should give you what you want (as longer as you can run the latest develop version of SCF).

**Comment by Flask** [comment]
Thank you for the clarification @jandubois!

Running the current develop version unfortunately is not feasible for us at the moment, but since it is in develop, could we expect to have it in one of the future releases?

**Comment by jandubois** [comment]
> could we expect to have it in one of the future releases

It landed too late to be in the SCF 2.14.x release series (CAP 1.3), but I would expect it to be available in the releases thereafter.

**Comment by Flask** [comment]
glad to hear that :)
